### ISSUE TITLE
Broken HTML output because of missing Content-Type header 

### ISSUE THREAD

**Issue by Presskopp** [GlotPress/gp-translation-helpers#182]
## Description
Opening a URL containing translation-helpers you get a not usable, weird looking page

## Step-by-step reproduction instructions
Open https://translate.wordpress.org/projects/wp/dev/admin/de/default/-get-translation-helpers/16087888-106856545/?nohc
(It doesn't matter if you leave out the query ?nohc btw)
See the result

## Expected behavior
Either a good looking page or no page at all (redirect, 404)

## Actual behavior
weird looking page

## Screenshots
follows

## Device information
browser doesn't matter

@dd32 mentioned on slack:
https://wordpress.slack.com/archives/C02RP50LK/p1689905945893709?thread_ts=1689885012.927269&cid=C02RP50LK
 "Looks like it’s just missing an appropriate Content-Type header, it’s similar to you accessing a wp-json/ or admin-ajax.php url directly."

**Comment by Presskopp** [comment]
![image](https://github.com/GlotPress/gp-translation-helpers/assets/18373563/1f53addb-2dc1-4698-b08f-630908fbb379)

**Comment by amieiro** [comment]
The correct URL to load this translation is https://translate.wordpress.org/projects/wp/dev/admin/de/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=16087888&filters%5Btranslation_id%5D=106896828

The URL with the `-get-translation-helpers` path in an AJAX endpoint implemented in this plugin to get the content for the `History`, `Other locales` and `Discussion` in an async way.


![image](https://github.com/GlotPress/gp-translation-helpers/assets/1667814/1d8debe4-831e-42a0-b36f-e9e3e42936b9)

![image](https://github.com/GlotPress/gp-translation-helpers/assets/1667814/48d69aca-c83b-4571-bfb2-7310f6f78a13)

So, basically, you are accessing an endpoint intended to be used by the plugin asynchronously, but not to be accessed directly from the browser.

**Comment by akirk** [comment]
@amieiro like Dion said, could we please change the Content-Type header?

> Looks like it’s just missing an appropriate Content-Type header, it’s similar to you accessing a wp-json/ or admin-ajax.php url directly

Also in your screenshot I can see that the response header `Content-Type` is `text/html`.
